### DISCUSSION TITLE
geckos.io with nginx

### DISCUSSION THREAD

**Discussion by HonzaKubita** [geckosio/geckos.io#295] [Help / Questions]
Hi I am building a project that is using geckos.io and on my server I use nginx. I am not that skilled in nginx so I have no idea how can I set up nginx together with geckos.io, the main headache is the upd port range. I need it even for developing the app because the app is using WebXR and for some reason WebXR requires you to use https and there is no way to disable it, so I need nginx to even just proxy traffic on my LAN to my mac where the web servers are running. Any help appreciated.

**Comment by yandeu** [comment]
Hi

Normally the UDP connection from geckos does not need to go through the reverse proxy. Just open the right ports to the server that hosts geckos. 
Only the signaling (normal http) of geckos needs to go through your nginx.

Btw, you can develop and test your WebXR apps locally. It should work on localhost or if you need to run it over your local network try to use [five-server](https://github.com/yandeu/five-server?tab=readme-ov-file#another-example) with the https feature. 

Edit:

Or if you use a server like express, just add a self signed certificate to test it locally. There is really no need to set up a nginx just to develop/test a WebXR+geckos app.
